pgwire: handle oid.T_unknown

otan opened the pull request:
Resolves #70027

Release note (sql change): T_unknown ParamaterTypeOIDs in the PostgreSQL
Frontend/Backend protocol are now correctly handled.

rafiss reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71971)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rafiss)

___
*[pkg/sql/pgwire/testdata/pgtest/unknown, line 26 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71971#-MmyME88B6Okji2LaG2a:-MmyME88B6Okji2LaG2b:b-tqxxho) ([raw file](https://github.com/cockroachdb/cockroach/blob/67e8c1185ce3d2c6a9742fea23b6b1b97e464a51/pkg/sql/pgwire/testdata/pgtest/unknown#L26)):*
> ```
> Parse {"Name": "s1", "Query": "INSERT INTO unknown_test VALUES($1, $2, $3)", "ParameterOIDs":[705, 705, 705]}
> Describe {"ObjectType": "S", "Name": "s1"}
> Bind {"DestinationPortal": "p1", "PreparedStatement": "s1", "ParameterFormatCodes": [0], "Parameters": [{"text":"1"}, {"text":"one"}, {"text":"1.5"}]}
> ```

Can we add a pgwire test to make sure we handle this in the binary protocol too?

Need to use `"ParameterFormatCodes": [1,1,1]`


<!-- Sent from Reviewable.io -->

otan committed:
pgwire: handle oid.T_unknown

Release note (sql change): T_unknown ParamaterTypeOIDs in the PostgreSQL
Frontend/Backend protocol are now correctly handled.

rafiss reviewed (approved):
lgtm!!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71971#-:-Mn1IWmLEvUT1CNPwzSM:b-nzoorr)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rafiss)

___
*[pkg/sql/pgwire/testdata/pgtest/unknown, line 26 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71971#-MmyME88B6Okji2LaG2a:-Mn1IQbr-i5jwn_H96IH:b-e918ht) ([raw file](https://github.com/cockroachdb/cockroach/blob/67e8c1185ce3d2c6a9742fea23b6b1b97e464a51/pkg/sql/pgwire/testdata/pgtest/unknown#L26)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

done, but i think all the magic is really at the "Parse" and "Describe" layer.
</blockquote></details>

yeah true - i think the binary test really just verifies that we handled it for both text+binary in `DecodeDatum`


<!-- Sent from Reviewable.io -->
